src/evaluator.ts

```diff
@@ -92,6 +105,72 @@ export class Evaluator {
     }
   }
 
+  private isTargetInList(target: Target, targets: Target[]): boolean {
+    return (
+      targets.find((elem) => elem.identifier === target.identifier) != undefined
+    );
+  }
+
+  private convertApiTargetsToEvalTargets(targets: ApiTarget[]): Target[] {
+    return targets.map(
+      (elem) =>
+        ({
+          identifier: elem.identifier,
+          name: elem.name,
+          anonymous: elem.anonymous,
+          attributes: elem.attributes,
+        } as Target),
+    );
+  }
+
+  private isTargetIncludedOrExcludedInSegment(
+    segments: string[],
+    target: Target,
+  ): boolean {
+    for (const segmentIdentifier of segments) {
+      const segment = this.query.getSegment(segmentIdentifier);
+
+      if (segment) {
+        // Should Target be excluded - if in excluded list we return false
+        if (
+          this.isTargetInList(
+            target,
+            this.convertApiTargetsToEvalTargets(segment.excluded),
+          )
+        ) {
+          // log.debug(
+          //   "Target %s excluded from segment %s via exclude list\n",
+          //   target.getName(), segment.getName());
+          return false;
+        }
+
+        // Should Target be included - if in included list we return false
+        if (
+          this.isTargetInList(
+            target,
+            this.convertApiTargetsToEvalTargets(segment.included),
+          )
+        ) {
+          // log.debug(
+          //   "Target %s included in segment %s via include list\n",
+          //   target.getName(), segment.getName());
+          return true;
+        }
+
+        // Should Target be included via segment rules
+        if (segment.rules) {
+          if (this.evaluateClauses(segment.rules, target)) {
+            // log.debug(
+            //   "Target %s included in segment %s via rules\n",
+            //   target.getName(), segment.getName());
+            return true;
+          }
+        }
```

Comment:
Might as well join this conditions in a single `if`
```suggestion
        if (segment.rules && this.evaluateClauses(segment.rules, target)) {
          // log.debug(
          //   "Target %s included in segment %s via rules\n",
          //   target.getName(), segment.getName());
          return true;
        }
```

Reply:
make sense